### PR TITLE
Adding state function to multiwalker

### PR DESCRIPTION
**Pull request by ffelten**
# Description

I added a state method giving a global view of the env on multiwalker

Realtes to #323 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [ ] I have run `pytest -v` and no errors are present.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by elliottower** [comment]
Awesome, have seen that issue up for a while thanks for the contribution Florian

File: pettingzoo/sisl/multiwalker/multiwalker.py [hunk 1]
```diff
@@ -160,6 +161,14 @@ def __init__(self, *args, **kwargs):
         # spaces
         self.action_spaces = dict(zip(self.agents, self.env.action_space))
         self.observation_spaces = dict(zip(self.agents, self.env.observation_space))
+        self.state_space = spaces.Box(
```
Comment (elliottower) [inline pettingzoo/sisl/multiwalker/multiwalker.py:164]: My only question is should this code be under multi walker base? The other two spaces are, but maybe it’s not possible

File: pettingzoo/sisl/multiwalker/multiwalker.py [hunk 2]
```diff
@@ -239,3 +248,17 @@ def step(self, action):
...
+        all_walker_obs = self.env.get_last_obs()
+        all_walker_obs = np.array(list(all_walker_obs.values())).flatten()
+        package_obs = np.array(
+            [
+                self.env.package.position.x,
+                self.env.package.position.y,
+                self.env.package.angle,
+            ]
+        )
+        global_state = np.concatenate((all_walker_obs, package_obs)).astype(np.float32)
```
Comment (elliottower) [inline pettingzoo/sisl/multiwalker/multiwalker.py:262]: Makes sense to me at least, the two things that define the state are the package and the different walkers observations

**Review by elliottower — APPROVED** [review]

**Review by jjshoots — CHANGES_REQUESTED** [review]
LGTM. But yeah I would honestly put this into the base env since `multiwalker.py`'s file structure basically mirrors other sisl environment's file structure.

**Comment by ffelten** [comment]
> LGTM. But yeah I would honestly put this into the base env since `multiwalker.py`'s file structure basically mirrors other sisl environment's file structure, it's already calling `self.env.x` so moving it into the base env makes more sense too.

@jjshoots done

**Review by jjshoots — APPROVED** [review]

**Comment by elliottower** [comment]
Looks good to me as well, cheers (and happy holidays)

**Pull request merged**